src/mentorship/features/history.py

```diff
@@ -0,0 +1,18 @@
+from typing import Optional, Tuple
+import pandas as pd
+
+
+def cut_history(
+    X: pd.DataFrame,
+    date_column: str,
+    keep_interval: pd.Timedelta,
+    y: Optional[pd.Series] = None
+) -> Tuple[pd.DataFrame, Optional[pd.Series]]:
+
+    last_date = pd.to_datetime(X[date_column].max())
+    X_train = X[X[date_column] >= str(last_date - keep_interval).split(' ')[0]]
```

Comment:
Not sure that string conversion is required here.
```suggestion
    X_train = X[X[date_column] >= last_date - keep_interval]
```

Reply:
I need this string conversion because >= operator not supported between 'str' and 'Timestamp' instances (date_column has 'str' type)

Reply:
Is your comparison safe or in other words, does comparison of date strings gives always give you the same result as the comparison of the original dates?

Reply:
It is safe in this situation, because dates are in format: yyyy-MM-dd, and string comparisons stop at the first difference and use character comparisons to determine order